app/controllers/searches_controller.rb

```diff
@@ -13,6 +13,12 @@ def show
     redirect_to rubygem_path(@exact_match) if @gems == [@exact_match]
   end
 
+  def autocomplete
+    render json: Rubygem.search(params[:query], es: true).map(&:name)
```

Comment:
Here I retrieve up to 30 records but show only 5 on UI. On my machine, there is no difference between 5 and 30 records so I leave the defaults.